### PR TITLE
Allow `impl ~const Trait` opaque types

### PR DESCRIPTION
**Pull request by fee1-dead**

### CODE REVIEW

File: src/test/ui/rfc-2632-const-trait-impl/const-impl-trait.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+struct S;
+impl const T for S {}
+
+const fn rpit() -> impl ~const T { S }
+
+const fn apit(_: impl ~const T + ~const Destruct) {}
+
+const fn rpit_assoc_bound() -> impl IntoIterator<Item: ~const T> { Some(S) }
+
+const fn apit_assoc_bound(_: impl IntoIterator<Item: ~const T> + ~const Destruct) {}
```
Comment (oli-obk) [inline src/test/ui/rfc-2632-const-trait-impl/const-impl-trait.rs:26]: please also test something like these (maybe returning an `impl PartialEq` and comparing at the call site?

Comment (fee1-dead) [reply src/test/ui/rfc-2632-const-trait-impl/const-impl-trait.rs:32]: done.

**Comment by oli-obk** [comment]
r=me with tidy fixed

**Comment by chriss0612** [comment]
That is definitely a feature that cleans up a lot of const code!
Does this work with rpitit?
In either case its probably worth adding a test for that.

**Pull request merged**